Gems/ROS2/Code/Include/ROS2/Frame/ROS2FrameSystemComponent.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/RTTI/RTTIMacros.h>
+#include <AzCore/std/containers/map.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/parallel/mutex.h>
+#include <AzCore/std/string/string.h>
+#include <ROS2/Frame/ROS2FrameSystemBus.h>
+
+namespace ROS2
+{
+    // Handler class for the ROS2FrameEditorComponent. It watches for changes in the entity tree and notifies about moves.
+    class ROS2FrameSystemTransformHandler : public AZ::TransformNotificationBus::Handler
+    {
+    public:
+        AZ_RTTI(ROS2FrameSystemTransformHandler, "{0b4324bb-4c36-4a86-8827-a044a6ac44e8}");
+
+        //! Override of the AZ::TransformNotificationBus.
+        void OnParentChanged(AZ::EntityId oldParent, AZ::EntityId newParent) override;
+
+        //! Add a frame entity which should be notified about a change in the tree.
+        //! @param frameEntityId frame to notify.
+        void AddFrameEntity(AZ::EntityId frameEntityId);
+
+        //! Remove a frame entity which shouldn't be notified about a change in the tree.
+        //! @param frameEntityId frame to remove.
+        void RemoveFrameEntity(AZ::EntityId frameEntityId);
+
+        //! Get size of the m_frameEntities.
```

Comment:
Change documentation here. "This returns size of member X" -> returns count of frames that has been added to the notification system. Or similar